packages/runtime/container-runtime/src/dataStores.ts

```diff
@@ -410,8 +441,14 @@ export class DataStores implements IDisposable {
 		localMessageMetadata: unknown,
 	) {
 		const envelope = message.contents as IEnvelope;
+
+		// If the data store has been deleted, log an error and ignore this message. This helps prevent document
+		// corruption in case a deleted data store accidentally submitted an op.
+		if (this.checkAndLogIfDeleted(envelope.address, "Changed", "processFluidDataStoreOp")) {
```

Comment:
I must have missed adding the check here.
I thought checking for deleted is mostly a performance optimization. The common case is that the context will exist so we should not spend time doing lookups for deleted.

Another question is what should the behavior be if the context exists but GC marked it as deleted? Ideally, that should never happen because GC first asks container runtime to delete it and then adds it to the deleted list. However, if it does, its probably fine to let the flow go through?

Reply:
Without the `context === undefined`, `checkAndLogIfDeleted` will assert if there's a context for it. So checking first protects that assert, in addition to the perf savings.

But on the flip side... that's where your question comes in.  Do we want to _always_ validate that Deleted ==> Undefined Context?  Aka Context Defined ==> Not Deleted.  Probably overkill.  Could probably even just remove the assert and move the `context === undefined` check to the top of `checkAndLogIfDeleted` (early return if Context is Defined).

Reply:
But maybe we're missing something, and would actually benefit from continuously asserting Context Defined ==> Not Deleted, at least for a time?  Let's just be consistent in how we express the assumptions / checks (could be a follow-up PR).

Reply:
I totally missed that there is an assert inside `checkAndLogIfDeleted` as well. I removed that assert and updated the logic to the following:
1. If the data store is deleted, the GC_Deleted_DataStore_* log will indicate whether it exists on the DataStore's list with an `exists` property. The operation will be ignored.
2. If the data store is not deleted but does not exist, it hits the regular assert that has nothing to do with GC.

The reason I removed the assert is because the point of this change is to not cause corruption in case of bugs where something is accidentally deleted. So, in that spirit, we should ignore this issue too.